client/src/ui/modules/navigation/TopMiddleNavigation.tsx

```diff
@@ -87,22 +82,21 @@ export const TopMiddleNavigation = () => {
     const url = new Position({ x: newPosition.x, y: newPosition.y }).toMapLocationUrl();
 
     setPreviewBuilding(null);
-    if (entityId) {
-      setRealmEntityId(entityId);
-    }
 
-    setLocation(url);
-    window.dispatchEvent(new Event("urlChanged"));
+    setLocationAndEmitEvent(url);
```

Comment:
better to have the function called `handleUrlChange` instead, an and statement in a function name implies a double responsibility, which we do not want